sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1689,7 +1689,10 @@ def _environment_to_df(environment: Environment) -> pd.DataFrame:
             {
                 "name": environment.name,
                 "snapshots": json.dumps(
-                    [snapshot.dict(mode="json") for snapshot in environment.snapshots]
+                    [
+                        s.dict(mode="json") if isinstance(s, SnapshotTableInfo) else s
```

Comment:
Can we somehow check the type only once?

Reply:
Moved upstream to environment class per our discussion.